app/routes/list-project.tsx

```diff
@@ -0,0 +1,531 @@
+import { Form, useActionData, useTransition } from "@remix-run/react";
+import { ChangeEvent, FC, useRef, useState } from "react";
+import ReactSelect, { MultiValue } from "react-select";
+import cx from "classnames";
+import Button from "~/components/Button";
+import TextArea from "~/components/TextArea";
+import TextField from "~/components/TextField";
+import {
+  ROLE_INTERESTS,
+  SUBJECT_INTERESTS,
+  TECH_INTERESTS,
+} from "~/utils/tags";
+import formStyles from "~/styles/forms.css";
+import markdownStyles from "~/styles/markdown.css";
+import TwitterIcon from "~/components/icons/TwitterIcon";
+import { LinkIcon, MarkGithubIcon } from "@primer/octicons-react";
+import ProjectPreview from "~/components/ProjectPreview";
+import { Project } from "~/types";
+import ProjectSubmissionSpinner from "~/components/ProjectSubmissionSpinner";
+import RequiredMarker from "~/components/RequiredMarker";
+import ExternalLink from "~/components/ExternalLink";
+import mapThumbnailSrc from "~/images/map_thumbnail.png";
+import { formatFileSize, maybeStringToArray } from "~/utils/formatting";
+import FieldError from "~/components/FieldError";
+import {
+  ActionFunction,
+  json,
+  LoaderFunction,
+  redirect,
+} from "@remix-run/node";
+import {
+  getAccessToken,
+  getCurrentUser,
+  getCurrentUserOrRedirect,
+} from "~/session.server";
+import {
+  MAX_AVATAR_SIZE_BYTES,
+  parseListProjectForm,
+} from "~/utils/project-submission";
+import { createNewPullRequest } from "~/github.server";
+import { getRepoOwnerAndName } from "~/utils/repo-url";
+
+export function links() {
+  return [
+    { rel: "stylesheet", href: formStyles },
+    { rel: "stylesheet", href: markdownStyles },
+  ];
+}
+
+/**
+ * Check that the user is logged in or redirect to the /login page
+ */
+export const loader: LoaderFunction = async ({ request }) => {
+  await getCurrentUserOrRedirect(request);
+  return json({});
+};
+
+/**
+ * This method is called when the form is submitted. It performs data validation
+ * and uploads files to GitHub before creating a PR. If all goes well, it
+ * redirects to a success page.
+ */
+export const action: ActionFunction = async ({ request }) => {
+  const currentUser = await getCurrentUser(request);
+  const accessToken = await getAccessToken(request);
+
+  if (!currentUser || !accessToken) {
+    throw new Error("You must be logged in to save your profile");
+  }
+
+  // TODO check whether we already have a project for this repository
+
+  // Parse the form
+  const { validationErrors, files, repoUrl } = await parseListProjectForm(
```

Comment:
There's a LOT of validation going on over in `parseListProjectForm()`. If there are validation errors, we'll send them back to the client. Otherwise, we'll send all that formatted data to GitHub.